### PR TITLE
LIMS-5404 : [Archiving][MainOrder]Make sure that if user cancel archive order,No order or suborders or analysis of the order will be archived

### PR DESCRIPTION
**Pull request by omnia353**
```

C:\Users\omnia\Documents\1limsNewMaster\1lims-automation>nosetests -vs --logging-level=WARNING ui_testing/testcases/basic_tests/test006_orders.py -m test090 --tc-file=config.ini
c:\users\omnia\lib\site-packages\nose\plugins\manager.py:394: RuntimeWarning: Unable to load plugin noseprogressive = noseprogressive:ProgressivePlugin: No module named 'fcntl'
  warn("Unable to load plugin %s: %s" % (ep, e),
2020-09-02 19:34:21.361 | INFO     | api_testing.apis.base_api:_get_authorized_session:46 - Get authorized api session.
2020-09-02 19:34:21.363 | INFO     | api_testing.apis.base_api:_get_authorized_session:47 - admin:admin
2020-09-02 19:34:22.091 | INFO     | api_testing.apis.base_api:_get_authorized_session:58 - session ID : QZtC0yceT3K-ogx4S9pFAMK3ERQWuxUxlpXsyyeGBn0 .....

DevTools listening on ws://127.0.0.1:56694/devtools/browser/b9576012-2841-4337-97bd-b67e5f780394
[Archiving][MainOrder]Make sure that if user cancel archive order, ...
2020-09-02 19:35:28.510 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test090_if_cancel_archive_order_no_order_suborder_analysis_will_archived
2020-09-02 19:35:34.646 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:577 - wait until page is loaded
2020-09-02 19:35:34.677 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:35:35.314 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:35:35.982 | INFO     | api_testing.apis.test_unit_api:set_name_configuration_name_only:427 - set test unit configuration
2020-09-02 19:35:36.496 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-02 19:35:36.502 | INFO     | api_testing.apis.orders_api:set_configuration:412 - set order configuration
2020-09-02 19:35:37.345 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-02 19:35:37.347 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test090_if_cancel_archive_order_no_order_suborder_analysis_will_archived:2881 - create order
2020-09-02 19:35:37.348 | INFO     | ui_testing.pages.order_page:create_new_order:124 -  Create new order.
2020-09-02 19:35:37.349 | INFO     | ui_testing.pages.orders_page:click_create_order_button:23 - Press create order button
2020-09-02 19:35:37.591 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:35:45.749 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-02 19:35:47.199 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-02 19:35:48.592 | INFO     | ui_testing.pages.order_page:set_new_order:20 - Set new order.
2020-09-02 19:35:56.020 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-02 19:36:03.166 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-02 19:36:10.425 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-02 19:36:19.313 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-02 19:36:26.372 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-02 19:36:34.107 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-02 19:36:40.985 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-02 19:36:42.481 | INFO     | ui_testing.pages.order_page:create_new_order:154 -  Order created with no : 93-2020 
2020-09-02 19:36:42.483 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test090_if_cancel_archive_order_no_order_suborder_analysis_will_archived:2883 - dupliacte the suborder for 2 times
2020-09-02 19:36:43.063 | INFO     | ui_testing.pages.base_pages:save:55 - save the changes
2020-09-02 19:36:43.066 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:36:43.916 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:36:45.848 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders/suborder/?orderId=23&deleted=0
2020-09-02 19:36:46.349 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-02 19:36:48.381 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:577 - wait until page is loaded
2020-09-02 19:36:48.405 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:36:49.247 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:36:50.181 | INFO     | ui_testing.pages.orders_page:filter_by_order_no:138 - Filter by order no. : 93-2020
2020-09-02 19:36:51.079 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:577 - wait until page is loaded
2020-09-02 19:36:51.247 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:36:52.048 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:36:52.957 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:36:53.758 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test090_if_cancel_archive_order_no_order_suborder_analysis_will_archived:2893 - click on arhcive then cancel popup
2020-09-02 19:36:53.760 | INFO     | ui_testing.pages.orders_page:archive_main_order_from_order_option:323 - archive main order from the order's active table
2020-09-02 19:36:53.785 | INFO     | ui_testing.pages.base_pages:open_row_options:280 - open record options menu
2020-09-02 19:36:53.847 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:36:54.724 | INFO     | ui_testing.pages.orders_page:archive_main_order_from_order_option:333 - Cancel the popup
2020-09-02 19:36:54.873 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test090_if_cancel_archive_order_no_order_suborder_analysis_will_archived:2897 - go to archived orders
2020-09-02 19:36:54.875 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:36:55.964 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-02 19:37:07.187 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-09-02 19:37:08.448 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok

----------------------------------------------------------------------
Ran 1 test in 168.202s

```

### CODE REVIEW

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 1]
```diff
@@ -2871,3 +2871,30 @@ def test084_filter_by_changed_by(self):
...
             self.orders_page.close_child_table(source=results[i])
+
+    def test090_if_cancel_archive_order_no_order_suborder_analysis_will_archived(self):
+        """
+        [Archiving][MainOrder]Make sure that if user cancel archive order,
+        No order or suborders or analysis of the order will be archived
+        LIMS-5404
+        """
+        self.info('create order')
+        self.order_page.create_new_order(material_type='Raw Material', save=False)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:2882]: Why do you create new order using GUI then duplicate it?  Is it required in jira  ticket? 
You can select random order and filter with order no then archive it

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 2]
```diff
@@ -2871,3 +2871,30 @@ def test084_filter_by_changed_by(self):
...
+        self.assertEqual(3, len(suborders))
+        self.orders_page.get_orders_page()
+        self.orders_page.filter_by_order_no(filter_text=order_no)
+        self.info('click on arhcive then cancel popup')
+        self.orders_page.archive_main_order_from_order_option(check_pop_up=True, confirm=False)
+        table_records = self.orders_page.result_table(element='general:table')
+        self.assertEqual(1, len(table_records) - 1)
+        self.info('go to archived orders')
+        self.orders_page.get_archived_items()
+        self.orders_page.search(order_no)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:2899]: Replace search with filter by order no.. As search gets all data with this no may be analysis has same no of your order.

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 3]
```diff
@@ -2871,3 +2871,30 @@ def test084_filter_by_changed_by(self):
...
+        self.assertEqual(len(self.order_page.result_table()) - 1, 0)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:2900]: As states in test case you need to assert that all analysis  of selected order not archived also.

**Review by ghada-37 — CHANGES_REQUESTED** [review]

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 1 above)
Comment (omnia353) [reply ui_testing/testcases/basic_tests/test006_orders.py:2882]: in the ticket it said in step 2 add order and in step 3 enter valid data

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 2 above)
Comment (omnia353) [reply ui_testing/testcases/basic_tests/test006_orders.py]: done

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 3 above)
Comment (omnia353) [reply ui_testing/testcases/basic_tests/test006_orders.py]: done

**Review by ghada-37 — APPROVED** [review]

**Pull request merged**